lib/inc/internal/config_delayed_merge.hpp

```diff
@@ -0,0 +1,21 @@
+#include <hocon/config_value.hpp>
+#include <internal/unmergeable.hpp>
+#include <vector>
+
+#pragma once
+
+namespace hocon {
+
+    class config_delayed_merge : public config_value, public unmergeable {
+    public:
+        config_delayed_merge(shared_origin origin, std::vector<shared_value> stack);
+
+        config_value_type value_type() const override;
+        std::vector<shared_value> unmerged_values() const override;
```

Comment:
I wonder if this (and thus the unmergeable interface) should return a `const&`. Not clear yet.